In [1]:
import pandas as pd
import numpy as np
# import pydot

import matplotlib.pyplot as plt
import matplotlib
import seaborn as sns
from sklearn.model_selection import train_test_split

import statsmodels.formula.api as smf
from sklearn.tree import DecisionTreeRegressor
from sklearn.ensemble import RandomForestRegressor
from sklearn.ensemble import GradientBoostingRegressor

from sklearn.metrics import mean_squared_error
from sklearn.metrics import mean_absolute_error

from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.ensemble import GradientBoostingClassifier
from sklearn.neural_network import MLPClassifier

from sklearn.svm import SVC

from sklearn.metrics import f1_score,roc_curve,auc,precision_score,recall_score

from sklearn.metrics import confusion_matrix

import seaborn as sns
from scipy import stats
import scipy.stats
import statsmodels.api as smd
from statsmodels.stats.proportion import proportions_ztest

# 가설검정, 분산분석하는 패키지
from statsmodels.formula.api import ols # 분산분석하기? 모델링?
from statsmodels.stats.anova import anova_lm # 아노바 데이터프레임 형태 출력
from scipy.stats import shapiro
%matplotlib inline
from sklearn.preprocessing import scale,minmax_scale,robust_scale
from scipy.stats import probplot
from sklearn.model_selection import train_test_split
from statsmodels.api import Logit

# font_path = 'C:/Windows/Fonts/gulim.ttc'
font_path = 'C:/Windows/Fonts/gulim.ttc'
font_name = matplotlib.font_manager.FontProperties(fname=font_path).get_name()
matplotlib.rc('font', family = font_name)

from matplotlib import font_manager,rc

from statsmodels.stats.outliers_influence import variance_inflation_factor
from sklearn.linear_model import LinearRegression
from sklearn.feature_selection import RFE
from sklearn.preprocessing import StandardScaler
from statsmodels.api import qqplot, add_constant

# from sklearn.tree import export_graphviz
# import graphviz

from sklearn.metrics import mean_squared_error
from sklearn.metrics import mean_absolute_error
import numpy as np
# import pydot

import matplotlib.pyplot as plt
import matplotlib
import seaborn as sns
from sklearn.model_selection import train_test_split

import statsmodels.formula.api as smf
from sklearn.tree import DecisionTreeRegressor
from sklearn.ensemble import RandomForestRegressor
from sklearn.ensemble import GradientBoostingRegressor

from sklearn.metrics import mean_squared_error
from sklearn.metrics import mean_absolute_error

from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.ensemble import GradientBoostingClassifier
from sklearn.neural_network import MLPClassifier

from sklearn.svm import SVC

from sklearn.metrics import f1_score,roc_curve,auc,precision_score,recall_score

from sklearn.metrics import confusion_matrix

import seaborn as sns
from scipy import stats
import scipy.stats
import statsmodels.api as smd
from statsmodels.stats.proportion import proportions_ztest

# 가설검정, 분산분석하는 패키지
from statsmodels.formula.api import ols # 분산분석하기? 모델링?
from statsmodels.stats.anova import anova_lm # 아노바 데이터프레임 형태 출력
from scipy.stats import shapiro
%matplotlib inline
from sklearn.preprocessing import scale,minmax_scale,robust_scale
from scipy.stats import probplot
from sklearn.model_selection import train_test_split
from statsmodels.api import Logit

# # font_path = 'C:/Windows/Fonts/gulim.ttc'
# font_path = '/usr/share/fonts/opentype/noto/NotoSerifCJK-ExtraLight.ttc'
# font_name = matplotlib.font_manager.FontProperties(fname=font_path).get_name()
# matplotlib.rc('font', family = font_name)

from matplotlib import font_manager,rc

from statsmodels.stats.outliers_influence import variance_inflation_factor
from sklearn.linear_model import LinearRegression
from sklearn.feature_selection import RFE
from sklearn.preprocessing import StandardScaler
from statsmodels.api import qqplot, add_constant

# from sklearn.tree import export_graphviz
# import graphviz

from sklearn.metrics import mean_squared_error
from sklearn.metrics import mean_absolute_error

In [2]:
df = pd.read_csv('insu_nh_h_screen.csv',encoding='euckr')

C:\Users\BHU\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3063: DtypeWarning: Columns (10,11,12,13,14,15,17,18,19,20,21) have mixed types.Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [3]:
np.random.seed(123)
sample = df.sample(frac=0.2)

In [4]:
test = sample.reset_index()
test.to_csv('raw_screen.csv',encoding='euckr')

In [5]:
data = sample[['bp_max','bp_min','bp_gluc','bt_mch','bt_chol','gender','smoke_flag','drinking_flag']]
ls = ['bp_max','bp_min','bp_gluc','bt_mch','bt_chol','gender','smoke_flag','drinking_flag']

In [6]:
for i in ls :
    data = data[data[i] != '.']
data

C:\Users\BHU\anaconda3\lib\site-packages\pandas\core\ops\array_ops.py:253: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  res_values = method(rvalues)


bp_max bp_min bp_gluc bt_mch bt_chol  gender smoke_flag drinking_flag
11565    119     73       89     15     197       1         1             0 
23186    109     61       84     15     229       1         1             0 
127414   134     80      90     17     196        1         1             0 
35119    132     83      91     15     180        1         3             1 
50103    102     60      93     12     155        2         1             0 
...       ...    ...     ...    ...     ...     ...        ...           ...
22859    115     75       98     13     190       2         1             0 
51068    125     75      95     16     198        1         3             1 
171938    120     70     93     15     189        1         2             0 
121319   123     71      82     15     196        1         2             1 
424      148     88      118     17     197       1         3             1 

[35421 rows x 8 columns]

In [7]:
data_2 = data.astype('float')

In [8]:
data_2.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 35421 entries, 11565 to 424
Data columns (total 8 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   bp_max         35421 non-null  float64
 1   bp_min         35421 non-null  float64
 2   bp_gluc        35421 non-null  float64
 3   bt_mch         35421 non-null  float64
 4   bt_chol        35421 non-null  float64
 5   gender         35421 non-null  float64
 6   smoke_flag     35421 non-null  float64
 7   drinking_flag  35421 non-null  float64
dtypes: float64(8)
memory usage: 2.4 MB


## 1분류 유의변수

'bt_chol','bp_judge','bt_gluc_judge','bt_mchc_judge','gender'

- bt_gluc : 100 미만 정상, 100~125 당뇨 전 단계
- mch : 남자 14-17.5 g/dL, 여자 12.3-15.3 g/dL
- 성별 : 남자 1 여자 2
- 혈압 : max 120 130mmHg, min 80 85

In [9]:
data_2['bp_judge'] = np.where((data_2['bp_max'].between(0,130)) & (data_2['bp_min'].between(0,85)),0,1)
data_2['bt_gluc_judge'] = np.where(data_2['bp_gluc']<100,0,1)

In [10]:
data_2.groupby('bt_gluc_judge').count()

bp_max  bp_min  bp_gluc  bt_mch  bt_chol  gender  smoke_flag  \
bt_gluc_judge                                                                 
0               22681   22681    22681   22681    22681   22681       22681   
1               12740   12740    12740   12740    12740   12740       12740   

               drinking_flag  bp_judge  
bt_gluc_judge                           
0                      22681     22681  
1                      12740     12740

In [11]:
data_m = data_2[data_2['gender'] == 1]
data_f = data_2[data_2['gender'] == 2]
data_m['bt_mchc_judge'] = np.where(data_m['bt_mch'].between(14,17.5),0,1)
data_f['bt_mchc_judge'] = np.where(data_f['bt_mch'].between(12.3,15.3),0,1)

data_3 = pd.concat([data_m, data_f])
data_3

C:\Users\BHU\anaconda3\lib\site-packages\ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  This is separate from the ipykernel package so we can avoid doing imports until
C:\Users\BHU\anaconda3\lib\site-packages\ipykernel_launcher.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  after removing the cwd from sys.path.


bp_max  bp_min  bp_gluc  bt_mch  bt_chol  gender  smoke_flag  \
11565    119.0    73.0     89.0    15.0    197.0     1.0         1.0   
23186    109.0    61.0     84.0    15.0    229.0     1.0         1.0   
127414   134.0    80.0     90.0    17.0    196.0     1.0         1.0   
35119    132.0    83.0     91.0    15.0    180.0     1.0         3.0   
141647   121.0    90.0     98.0    15.0    212.0     1.0         2.0   
...        ...     ...      ...     ...      ...     ...         ...   
161781   118.0    85.0     88.0    13.0    195.0     2.0         1.0   
174466   105.0    65.0    103.0    13.0    224.0     2.0         1.0   
122883   120.0    84.0    103.0    13.0    194.0     2.0         1.0   
107700   121.0    70.0     89.0    13.0    241.0     2.0         1.0   
22859    115.0    75.0     98.0    13.0    190.0     2.0         1.0   

        drinking_flag  bp_judge  bt_gluc_judge  bt_mchc_judge  
11565             0.0         0              0              0  
23186             0.0         0              0              0  
127414            0.0         1              0              0  
35119             1.0         1              0              0  
141647            0.0         1              0              0  
...               ...       ...            ...            ...  
161781            0.0         0              0              0  
174466            0.0         0              1              0  
122883            0.0         0              1              0  
107700            0.0         0              0              0  
22859             0.0         0              0              0  

[35421 rows x 11 columns]

In [12]:
data_3.groupby('bt_mchc_judge').count()

bp_max  bp_min  bp_gluc  bt_mch  bt_chol  gender  smoke_flag  \
bt_mchc_judge                                                                 
0               30219   30219    30219   30219    30219   30219       30219   
1                5202    5202     5202    5202     5202    5202        5202   

               drinking_flag  bp_judge  bt_gluc_judge  
bt_mchc_judge                                          
0                      30219     30219          30219  
1                       5202      5202           5202

In [13]:
data_3 = data_3.reset_index()
data_3

index  bp_max  bp_min  bp_gluc  bt_mch  bt_chol  gender  smoke_flag  \
0       11565   119.0    73.0     89.0    15.0    197.0     1.0         1.0   
1       23186   109.0    61.0     84.0    15.0    229.0     1.0         1.0   
2      127414   134.0    80.0     90.0    17.0    196.0     1.0         1.0   
3       35119   132.0    83.0     91.0    15.0    180.0     1.0         3.0   
4      141647   121.0    90.0     98.0    15.0    212.0     1.0         2.0   
...       ...     ...     ...      ...     ...      ...     ...         ...   
35416  161781   118.0    85.0     88.0    13.0    195.0     2.0         1.0   
35417  174466   105.0    65.0    103.0    13.0    224.0     2.0         1.0   
35418  122883   120.0    84.0    103.0    13.0    194.0     2.0         1.0   
35419  107700   121.0    70.0     89.0    13.0    241.0     2.0         1.0   
35420   22859   115.0    75.0     98.0    13.0    190.0     2.0         1.0   

       drinking_flag  bp_judge  bt_gluc_judge  bt_mchc_judge  
0                0.0         0              0              0  
1                0.0         0              0              0  
2                0.0         1              0              0  
3                1.0         1              0              0  
4                0.0         1              0              0  
...              ...       ...            ...            ...  
35416            0.0         0              0              0  
35417            0.0         0              1              0  
35418            0.0         0              1              0  
35419            0.0         0              0              0  
35420            0.0         0              0              0  

[35421 rows x 12 columns]

In [14]:
# 1분류 스크린 데이터 내보내기_35000개 데이터 샘플링
data_3.to_csv('screen_1.csv',encoding='euckr')

## 2분류 유의변수

'age', 'bp_max','bp_judge','bt_gluc_judge','bt_mchc_judge','bt_hbsa'
-> hbsa 제외 진행

In [138]:
result = pd.read_csv('result_1.csv',encoding='euckr')
join_1 = result[result['risk'] == 1]

In [140]:
sample_2 = sample.reset_index()

In [141]:
join_2 = sample_2[['age_group','bp_max','index']]

In [145]:
test_2 = pd.merge(join_1,join_2,on='index',how='left').drop('Unnamed: 0',axis=1)

In [171]:
age_t = test_2['age_group'].str.split('~')

In [187]:
ls_age = []
for i in age_t :
    if i != ['85세+'] :
        k = (int(i[0]) + int(i[1]))/2
        ls_age.append(k)
    else :
        ls_age.append(87)

test_2['age'] = ls_age
test_2.to_csv('screen_2.csv',encoding='euckr')